Port 508 compliance changes from cBioPortal PRs

fedde-s opened the pull request:
Applying Sander’s changes from cBioPortal/cbioportal#1219 (and cBioPortal/cbioportal#1177, in progress) to the corresponding locations in the source code of muation-mapper

fedde-s committed:
Port 508 compliance changes from cBioPortal PR

Applying Sander's changes from cBioPortal/cbioportal#1219 to the
corresponding locations in the source code of muation-mapper

fedde-s commented:
This is the pull request announced to resolve issue #37. @onursumer, could you review the changes in their new context?

fedde-s commented:
I have not updated the corresponding files in the build/ directory even though they are tracked with Git, as they seemed to have already been out-of-date with other changes.

onursumer commented:
Thanks @fedde-s! Don't worry about the build files, I will update them once we merge rc branch into master.

onursumer commented:
@fedde-s: Looks like we still need to add `alt` property to `img` tags, right?

onursumer commented on src/js/controller/MutationDetailsController.js:
```diff
@@ -324,6 +324,17 @@ function MutationDetailsController(
 			new Mutation3dController(mutationDetailsView, mainView,
 				_mut3dVisView, view3d, pdbProxy, mutationUtil, gene);
 		}
+
```
It might be better to move this code segment into a view component, preferably into the corresponding view where we actually initialize the table. This may not be safe, since component initialization might be asynchronous depending on how we retrieve the data.

fedde-s commented:
@onursumer: Ah, I see, the changes from pull request cBioPortal/cbioportal#1177. I’ll look into those by Monday. I can add that to the same PR if you haven’t merged by then or simply create a second one—whichever you prefer.

onursumer commented:
@fedde-s You can update this PR, thanks!

fedde-s commented:
@onursumer: I haven’t tried deploying the release candidate of mutation-mapper to one of my dev installations of cBioPortal for testing it, as I am unsure how to. Is it documented somewhere, or as simple as copying the `build/` files somewhere? It might help me to see whether the issues still exist in the release candidate, as I am not yet strongly familiar with this part of the code.

fedde-s committed:
Add alt tags to images

Porting Sander’s changes from cBioPortal pull request
cBioPortal/cbioportal#1177 to the corresponding locations in the source
code of mutation-mapper

onursumer commented:
@fedde-s : sorry for the late reply. To test mutation mapper with portal RC, I build the project with `gulp make`, and then manually copy the build files `mutationMapper.js`, `mutationMapperTemplates.html`, and `mutationMapper.min.css` over the corresponding locations in portal project. Currently we have a minimal readme: https://github.com/cBioPortal/mutation-mapper/blob/master/README.md

I think we need a better/automated integration of the mutation mapper library into portal. Probably with the help of a package management tool like **bower**.

onursumer commented on src/js/controller/MutationDetailsController.js:
```diff
@@ -324,6 +324,17 @@ function MutationDetailsController(
+		// 508 compliance: add a title to each of the checkboxes provided by
+		// the ColVis library. As the offending checkboxes don't become visible
+		// until the button is clicked, bind it to the click event
+		$(".ColVis_MasterButton").click(function() {
```
@fedde-s : can you please move this part into `MutationDetailsTable`'s `fnInitComplete` function for now?

onursumer commented:
@fedde-s : I am merging this PR. I can move the `ColVis_MasterButton` related part after merge. Thanks!

fedde-s commented:
Ah, yes, sorry, busy day. And thanks for the hint about moving the ColVis fix: I’d ported it from another patch to what seemed to be the corresponding location in the new version, but the context of the code had changed a lot.

onursumer commented:
No problem! I already merged the PR and moved that part into `MutationDetailsTable`

pieterlukasse commented:
@onursumer can you add the PR reference where this code got submitted to cbioportal/cbioportal? Then we can track the status of this code in the main project.

onursumer commented:
@pieterlukasse https://github.com/cBioPortal/cbioportal/pull/1325 is the PR containing all the changes added with this PR.
